### PR TITLE
allow transpiler to provide alternative names for dependencies

### PR DESCRIPTION
**Pull request by vladima**
this functionality is used by external bundlers

### CODE REVIEW

**Comment by guybedford** [comment]
:+1: this makes the SystemJS builder typescript tests all pass for me now.

**Comment by vladima** [comment]
Pinging @mhegazy

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -5343,13 +5343,26 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
...
+                return undefined;
+            }
 
             function emitRequire(moduleName: Expression) {
                 if (moduleName.kind === SyntaxKind.StringLiteral) {
                     write("require(");
-                    emitStart(moduleName);
-                    emitLiteral(<LiteralExpression>moduleName);
-                    emitEnd(moduleName);
+                    let text = tryRenameExternalModule(<LiteralExpression>moduleName);
```
Comment (mhegazy) [inline src/compiler/emitter.ts]: can you add a comment, either here or on tryRenameExternalModule

**Comment by mhegazy** [comment]
:+1:

File: src/compiler/emitter.ts (hunk 1 above)
Comment (vladima) [inline src/compiler/emitter.ts]: done, added one on `tryRenameExternalModule`

**Pull request merged**